collector/system_collector.go

```diff
@@ -127,17 +236,45 @@ func (sc *systemCollector) collectClusterNodeMetrics() (systemStatusMetrics []sy
 		}
 		systemStatusMetrics = append(systemStatusMetrics, controllerStatusMetric)
 	}
+	return
+}
 
-	for _, m := range clusterNodesStatus.ManagementCluster {
-		managementStatusMetric := systemStatusMetric{
+func (sc *systemCollector) extractManagementNodeMetrics(clusterNodes administration.ClustersAggregateInfo) (systemStatusMetrics []systemStatusMetric) {
+	for _, m := range clusterNodes.ManagementCluster {
+		managementNodeMetric := systemStatusMetric{
 			IPAddress: m.RoleConfig.MgmtPlaneListenAddr.IpAddress,
 			Type:      "management",
 			Status:    0.0,
 		}
 		if strings.ToUpper(m.NodeStatus.MgmtClusterStatus.MgmtClusterStatus) == "CONNECTED" {
-			managementStatusMetric.Status = 1.0
+			managementNodeMetric.Status = 1.0
+		}
+		for _, prop := range m.NodeStatusProperties {
+			managementNodeMetric.CPUCores = float64(prop.CpuCores)
+
+			managementNodeMetric.LoadAverage = make([]float64, 3)
+			managementNodeMetric.LoadAverage[0] = float64(prop.LoadAverage[0])
+			managementNodeMetric.LoadAverage[1] = float64(prop.LoadAverage[1])
+			managementNodeMetric.LoadAverage[2] = float64(prop.LoadAverage[2])
+
+			managementNodeMetric.MemoryUse = float64(prop.MemUsed)
+			managementNodeMetric.MemoryTotal = float64(prop.MemTotal)
+			managementNodeMetric.MemoryCached = float64(prop.MemCache)
+
+			managementNodeMetric.SwapUse = float64(prop.SwapUsed)
+			managementNodeMetric.SwapTotal = float64(prop.SwapTotal)
+
+			managementNodeMetric.DiskUse = make(map[string]float64, 10)
+			managementNodeMetric.DiskTotal = make(map[string]float64, 10)
+			for _, disk := range prop.FileSystems {
+				managementNodeMetric.DiskUse[disk.Mount] = float64(disk.Used)
+				managementNodeMetric.DiskTotal[disk.Mount] = float64(disk.Total)
+			}
+
+			// Only process the first series of properties
```

Comment:
Can we use m.NodeStatusProperties[0], then extract `0` to constant.
We'll able to use the constant name to explain what is the first index actually mean